src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -214,6 +214,16 @@ public DiagnosticBuilderInternal(TextSpan textSpan)
                 "BCP038",
                 $"The property '{property}' is not allowed on objects of type {type}. Permissible properties include '{validUnspecifiedProperties.ConcatString("', '")}'.");
 
+            public ErrorDiagnostic DisallowedUnknownKeyProperty(object type) => new ErrorDiagnostic(
+                TextSpan,
+                "BCP039",
+                $"Additional properties are not allowed on objects of type {type}.");
+
+            public ErrorDiagnostic DisallowedUnknownKeyPropertyWithPermissibleProperties(object type, IEnumerable<string> validUnspecifiedProperties) => new ErrorDiagnostic(
+                TextSpan,
+                "BCP040",
+                $"Additional properties are not allowed on objects of type {type}. Permissible properties include '{validUnspecifiedProperties.ConcatString("', '")}'.");
```

Comment:
>Additional [](start = 18, length = 10)

My initial thought when reading this was "how is this different from BCP038?". Should we actually mention string interpolation in the error? Permissible properties would also be confusing because wouldn't allow interpolation, right?

Reply:
Good point.